### PR TITLE
Add repo-review

### LINKED ISSUE #448: adopt repo-review (with initial config)

### PR DESCRIPTION
**Pull request by ukmo-ccbunney**
## 🚀 Pull Request
Adds [repo-review](https://github.com/scientific-python/repo-review) checker as a pre-commit task 

### Description
Uses the [sp-repo-review](https://pypi.org/project/sp-repo-review/) plugin to run repository checks based on the [Scientific Python Developer Guide](https://learn.scientific-python.org/development)

- fixes: #448

### CODE REVIEW

**Comment by ukmo-ccbunney** [comment]
`ci-repo-review` task currently (correctly) failing with errors from `sp-repo-review` checker. Example output:
https://github.com/ukmo-ccbunney/cf-units/actions/runs/10906871427/attempts/1#summary-30269150465

**Comment by bjlittle** [comment]
Have you considered using the pre-commit hook instead?

This would get run by the `pre-commit.ci` service, but also force contributors to catch non-compliances prior to pushing the pull-request (if they install the pre-commit hooks).

This is the approach I've taken on GeoVista, see [here](https://github.com/bjlittle/geovista/blob/main/.pre-commit-config.yaml#L106-L111).

I think it's a slicker workflow 👍

**Comment by ukmo-ccbunney** [comment]
> Have you considered using the pre-commit hook instead?
> 
> This would get run by the `pre-commit.ci` service, but also force contributors to catch non-compliances prior to pushing the pull-request (if they install the pre-commit hooks).
> 
> This is the approach I've taken on GeoVista, see [here](https://github.com/bjlittle/geovista/blob/main/.pre-commit-config.yaml#L106-L111).
> 
> I think it's a slicker workflow 👍

Thanks @bjlittle - I'll checkout the pre-commit approach.

**Comment by ukmo-ccbunney** [comment]
Modified this to hook in `sp-repo-review` as a `pre-commit` task, rather than a GHA.

`sp-repo-review` reports plenty of failures (expected), so I have added these to the `ignore` list for now in the `pyproject.toml`. Assumedly, we will want to raise separate Issues for those we want to address.

File: pyproject.toml [hunk 1]
```diff
@@ -73,3 +73,26 @@ line_length = 79
 profile = "black"
 skip_gitignore = "True"
 verbose = "False"
+
+[tool.repo-review]
+# These are a list of the currently failing tests:
```
Comment (HGWright) [inline pyproject.toml:78]: Thanks @ukmo-ccbunney for your first CF-Units PR! I'm happy with the changes, only thing I could ask of you is to create an issue with a tasklist to consider/address the currently failing repo-review tests. (similar tasklist to this one scitools/iris#5780)

**Review by HGWright — CHANGES_REQUESTED** [review]
Just the one thing to do and then I'm happy to merge this in!

File: pyproject.toml (hunk 1 above)
Comment (ukmo-ccbunney) [reply pyproject.toml:78]: Done! See https://github.com/SciTools/cf-units/issues/462

**Review by HGWright — APPROVED** [review]
LGTM, thanks @ukmo-ccbunney!

**Comment by bjlittle** [comment]
Bingo! Nice one @ukmo-ccbunney :fire::rocket:

**Pull request merged**